lnd_services.go

```diff
@@ -414,6 +414,22 @@ func NewLndServices(cfg *LndServicesConfig) (*GrpcLndServices, error) {
 	return services, nil
 }
 
+// WithMacaroonAuth modifies the passed context to include the macaroon KV
+// metadata for the target Service. This method can be used to add the macaroon
+// at call time, rather than when the connection to the gRPC server is created.
+func (s *LndServices) WithMacaroonAuthForService(ctx context.Context,
+	service LnrpcServiceMac) (context.Context, error) {
+
+	var mac serializedMacaroon
+	var ok bool
+
+	if mac, ok = s.macaroons[service]; !ok {
```

Comment:
nit: could use shorter version:
```
mac, ok := s.macaroons[service]
if !ok {
...
}
```

Reply:
Thanks, I seem to forget that pattern a lot